sql/core/src/test/resources/sql-tests/inputs/datetime.sql

```diff
@@ -2,13 +2,16 @@
 
 -- [SPARK-31710] TIMESTAMP_SECONDS, TIMESTAMP_MILLISECONDS and TIMESTAMP_MICROSECONDS to timestamp transfer
 select TIMESTAMP_SECONDS(1230219000),TIMESTAMP_SECONDS(-1230219000),TIMESTAMP_SECONDS(null);
+select TIMESTAMP_SECONDS(1.23), TIMESTAMP_SECONDS(1.23d);
 select TIMESTAMP_MILLIS(1230219000123),TIMESTAMP_MILLIS(-1230219000123),TIMESTAMP_MILLIS(null);
 select TIMESTAMP_MICROS(1230219000123123),TIMESTAMP_MICROS(-1230219000123123),TIMESTAMP_MICROS(null);
--- overflow exception:
+-- overflow exception
 select TIMESTAMP_SECONDS(1230219000123123);
 select TIMESTAMP_SECONDS(-1230219000123123);
 select TIMESTAMP_MILLIS(92233720368547758);
 select TIMESTAMP_MILLIS(-92233720368547758);
+-- truncate exception
+select TIMESTAMP_SECONDS(0.1234567);
```

Comment:
Shall we have a test case for `allow truncate` together because this PR allows truncation for `double` type?

Reply:
This PR aims to allow truncation for both ANSI and legacy mode. Did I understand correctly?

Reply:
yes, float/double are approximate values and truncation should be always allowed.